include/geom/bounding_box.h

```diff
@@ -140,6 +140,13 @@ class BoundingBox : public std::pair<Point, Point>
    * d <= 0 if the point is inside the Bounding Box
    */
   Real signed_distance(const Point & p) const;
+
+  /**
+   * Scales each dimension of the bounding box by \p factor.
+   *
+   * Has no effect for dimensions that are invalid.
```

Comment:
Maybe replace "invalid" by "at numeric_limits::max in either direction"?  That's still a perfectly valid bounding box, it's just not one we can sensibly apply a relative scaling to in that dimension.